contracts/pods/MetadataRegistry.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+import {CoreRef} from "../refs/CoreRef.sol";
+
+/// @title Metadata registry for Pods
+/// @notice Exposes a single public method which should be called as part of a Pods proposal.
+/// @dev Expected that a call is made to this MetadataRegistry contract as the first function call in the
+/// calldata that is
+contract MetadataRegistry is CoreRef {
+    /// @notice Mapping identifying whether a particular proposal metadata was submitted for registration
+    /// @dev Maps the hash of the proposal metadata to a bool identifying if it was submitted
+    mapping(bytes32 => bool) public registration;
+
+    /// @notice Event that logs the metadata associated with a pod proposal
+    event RegisterProposal(
+        uint256 indexed podId,
+        uint256 indexed proposalId,
+        string metadata
+    );
+
+    constructor(address _core) CoreRef(_core) {}
+
+    /// @notice Get whether a pod proposal has been registered
+    /// @param podId Unique identifier of the pod for which metadata is being registered
+    /// @param proposalId Unique identifier of the proposal for which metadata is being registered
+    /// @param metadata Proposal metadata
+    function isProposalRegistered(
+        uint256 podId,
+        uint256 proposalId,
+        string memory metadata
+    ) external view returns (bool) {
+        bytes32 proposalHash = keccak256(
+            abi.encode(podId, proposalId, metadata)
+        );
+        return registration[proposalHash];
+    }
+
+    /// @notice Register a pod proposal. Specifically used as a layer ontop of a Gnosis safe to emit
+    ///         proposal metadata
+    /// @param podId Unique identifier of the pod for which metadata is being registered
+    /// @param proposalId Unique identifier of the proposal for which metadata is being registered
+    /// @param metadata Proposal metadata
+    // TODO: Restrict to only being callable by pods, to limit DDOS potential. Add new role
+    function registerProposal(
+        uint256 podId,
+        uint256 proposalId,
+        string memory metadata
+    ) external {
```

Comment:
Agree that this function needs to be permissioned, otherwise we are going to get spammed with garbage data.

Reply:
I'd take it a step further and possibly enforce that this is done atomically with pod creation. What are the pros and cons of this?

Reply:
Not sure I follow? My plan is to grant a new role `POD_ROLE`, who's admin is the `ROLE_ADMIN`, when the pod is deployed. 

I will then permission this `registerProposal()` function to that `POD_ROLE`